tests/unit/test_calibration.py

```diff
@@ -0,0 +1,271 @@
+    )
+    calibrator = calibration.Calibration()
+
+    sampling_rate = calibrator._get_sampling_rate(dummy_measure)
+
+    assert sampling_rate == 60
+
+
+def test_extract_no_motion_value_error() -> None:
+    """Check extract no motion."""
+    date = datetime.now()
+    delta = timedelta(seconds=1)
+    dummy_datetime = [date + (i * delta) for i in range(60)]
+    dummy_data = np.random.uniform(-0.29, 0.29, (60, 3))
+    dummy_measure = models.Measurement(
+        measurements=dummy_data, time=pl.Series(dummy_datetime).alias("time")
+    )
+    calibrator = calibration.Calibration()
+
+    with pytest.raises(
+        ValueError, match="Zero non-motion epochs found. Data did not meet criteria."
+    ):
+        calibrator._extract_no_motion(acceleration=dummy_measure)
+
+
+def test_extract_no_motion_sphere_error() -> None:
+    """Check extract no motion."""
+    dummy_measure = create_dummy_measurement(
+        sampling_rate=1, duration_hours=1, all_same_num=0.29
+    )
+    calibrator = calibration.Calibration()
+
+    with pytest.raises(calibration.SphereCriteriaError):
+        calibrator._extract_no_motion(acceleration=dummy_measure)
+
+
+def test_extract_no_motion() -> None:
+    """Check extract no motion."""
+    date = datetime.now()
+    delta = timedelta(seconds=1)
+    dummy_datetime = [date + (i * delta) for i in range(60)]
+    sphere_check_data = np.full((60, 3), 0.31)
+    sphere_check_data[30:] = -sphere_check_data[30:]
+
+    dummy_measure = models.Measurement(
+        measurements=sphere_check_data, time=pl.Series(dummy_datetime).alias("time")
+    )
+    calibrator = calibration.Calibration()
+
+    no_motion_data = calibrator._extract_no_motion(dummy_measure)
+
+    assert no_motion_data.shape[0] > 0, "No non-motion epochs found"
+    assert np.all(no_motion_data.min(axis=0) <= -calibrator.min_acceleration)
+    assert np.all(no_motion_data.max(axis=0) >= calibrator.min_acceleration)
+
+
+def test_closest_point_fit() -> None:
+    """Test closest point fit."""
+    dummy_no_motion = np.full((100, 3), (1 / np.sqrt(3)))
+    calibrator = calibration.Calibration()
+
+    linear_transform = calibrator._closest_point_fit(dummy_no_motion)
+
+    assert np.allclose(linear_transform["scale"], 1.0)
+    assert np.allclose(linear_transform["offset"], 0.0)
+
+
+def test_calibrate_calibration_error() -> None:
+    """Test failure due to calibration error."""
+    no_motion_data = np.full((3600, 3), 1.0)
+    dummy_measure = create_dummy_measurement(
+        sampling_rate=1, duration_hours=1, all_same_num=1.0
+    )
+    calibrator = calibration.Calibration(min_error=0.0001, max_iterations=5)
+
+    with mock.patch.object(
+        calibration.Calibration, "_extract_no_motion"
+    ) as mock_extracted_data:
+        mock_extracted_data.return_value = no_motion_data
+
+        with pytest.raises(calibration.CalibrationError):
+            calibrator._calibrate(dummy_measure)
```

Comment:
in a sense yes, the idea was to make the min_error and the max_iterations obviously extremely low it could never calibrate enough, and would fail due not being able to get the error down enough.